mycore-base/src/main/java/org/mycore/resource/provider/MCRResourceProvider.java

```diff
@@ -102,30 +110,57 @@ public BaseDirPrefixStripper(File baseDir) {
         }
 
         @Override
-        public String doStrip(String value) {
+        public List<String> getStrippedPaths(String value) {
             if (value.startsWith(prefix)) {
-                return value.substring(prefix.length());
+                return List.of(value.substring(prefix.length()));
             }
-            return null;
+            return Collections.emptyList();
         }
 
         @Override
         public String toString() {
             return prefix;
         }
 
-        public static Set<PrefixStripper> ofClassLoader(ClassLoader classLoader) {
-            Set<PrefixStripper> strippers = new LinkedHashSet<>();
-            List<URI> classpath = new ClassGraph().overrideClassLoaders(classLoader).getClasspathURIs();
-            classpath.forEach(uri -> {
-                if (uri.getScheme().equals("file")) {
-                    File file = new File(uri.getPath());
-                    if (file.isDirectory()) {
-                        strippers.add(new BaseDirPrefixStripper(file));
+    }
+
+    final class ClassLoaderPrefixStripper extends PrefixStripperBase {
+
+        private static final int CLASS_GRAPH_THREADS = 4;
+
+        private static final ExecutorService EXECUTOR_SERVICE = new AutoCloseableExecutorService(CLASS_GRAPH_THREADS);
+
+        private final ClassLoader classLoader;
+
+        public ClassLoaderPrefixStripper(ClassLoader classLoader) {
+            this.classLoader = Objects.requireNonNull(classLoader);
+        }
+
+        @Override
+        public List<String> getStrippedPaths(String value) {
+            List<String> potentialPaths = new LinkedList<>();
+            ClassGraph classGraph = new ClassGraph();
+            classGraph.overrideClassLoaders(classLoader);
+            try (ScanResult scanResult = classGraph.scan(EXECUTOR_SERVICE, CLASS_GRAPH_THREADS)) {
```

Comment:
This looks weird to me. Using a static EXECUTOR_SERVICE which is autoclosable. Doesn't this imply that once it's used, it is closed and therefore not avaiable anymore in other instances of ClassLoaderPrefixStripper? Or do I see that wrong?

Reply:
The `ExecutorService` is not used/closed in the  try-with-resourcse-statement, the `ScanResult` is. Only the `ScanResult` gets closed.

I keept using the `AutoCloseableExecutorService` while deconstructing the formally used method `ClassGraph#getClasspathURIs`, but the auto-closing feature isn't needed now. A simple `ThreadPoolExecutor` should suffice. Will change.

Reply:
ah ok, thanks for clarification.